turbinia/api/api_server.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Turbinia API server."""
+
+import io
+import logging
+import os
+import uvicorn
+import yaml
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.routing import APIRoute
+from fastapi.responses import Response
+from fastapi.staticfiles import StaticFiles
+
+from starlette_oauth2_api import AuthenticateMiddleware
+
+from turbinia import config
+from turbinia.api.routes.router import router
+
+log = logging.getLogger('turbinia:api_server')
+
+
+def get_application():
+  """Returns a FastAPI application object."""
+  description = '''This is Turbinia's API server description.'''
+  _app = FastAPI(
+      title='Turbinia API Server', description=description, version='1.0.0',
+      license_info={
+          'name': 'Apache License 2.0',
+          'url': 'https://www.apache.org/licenses/LICENSE-2.0.html'
+      }, routes=router.routes)
+  return _app
+
+
+def set_operation_ids(app):
+  """Simplify operation IDs so that generated API clients have
+    simpler function names.
+    Should be called only after all routes have been added.
+    """
+  for route in app.routes:
+    if isinstance(route, APIRoute):
+      route.operation_id = route.name
+
+
+def serve_static_content(app):
+  """Configure the application to serve static content.
+
+  This method must be called after all routes have been initialized.
+  """
+  root_content_path = os.path.realpath('../../web')
+  css_content_path = os.path.join(root_content_path, 'css')
+  js_content_path = os.path.join(root_content_path, 'js')
+
+  app.mount("/", StaticFiles(directory=root_content_path, html=True), name="/")
+  app.mount("/css", StaticFiles(directory=css_content_path), name="/css")
+  app.mount("/js", StaticFiles(directory=js_content_path), name="/js")
+
+
+def configure_authentication_providers(app):
+  """Configure the application's authentication providers.
+
+  Example provider configuration using starlette-oauth2-pai:
+
+  app.add_middleware(
```

Comment:
This looks good. oauth2-proxy does not support 2 client IDs at the moment, do you have a bug we can track so we know when it does support multiple webids?

Reply:
Correct. I'll create a bug fro tracking that feature in oauth2-proxy

Reply:
See https://github.com/oauth2-proxy/oauth2-proxy/issues/1740